apps/system/js/core.js

```diff
@@ -50,6 +54,80 @@
     },
 
     _start: function() {
+      ScreenManager && ScreenManager.turnScreenOn();
+      // We need to be sure to get the focus in order to wake up the screen
+      // if the phone goes to sleep before any user interaction.
+      // Apparently it works because no other window
+      // has the focus at this point.
+      window.focus();
+      // With all important event handlers in place, we can now notify
+      // Gecko that we're ready for certain system services to send us
+      // messages (e.g. the radio).
+      // Note that shell.js starts listen for the mozContentEvent event at
+      // mozbrowserloadstart, which sometimes does not happen till
+      // window.onload.
+      this.publish('mozContentEvent', {
+        type: 'system-message-listener-ready'
+      }, true);
+
+      this.loadWhenIdle([
+        'CameraTrigger',
+        'NotificationScreen',
+        'AirplaneMode',
+        'NotificationsSystemMessage',
+        'Accessibility',
+        'AlarmMonitor',
+        'DebuggingMonitor',
+        'TimeCore',
+        'GeolocationCore',
+        'TetheringMonitor',
+        'UsbCore',
+        'TextSelectionDialog',
+        'ExternalStorageMonitor',
+        'DeviceStorageWatcher',
+        'AppUsageMetrics',
+        'CellBroadcastSystem', // Blocked by integration test.
+        'CpuManager',
+        'HomeGesture',
+        'SourceView',
+        'TtlView',
+        'MediaRecording',
+        'QuickSettings',
+        'UsbStorage',
+        'MobileIdManager',
+        'FindmydeviceLauncher',
+        'FxAccountsManager',
+        'FxAccountsUI',
+        'NetworkActivity',
+        'CrashReporter',
+        'Screenshot',
+        'SoundManager',
+        'CustomDialogService',
+        'CarrierInfoNotifier'
+        // XXX: We should move this into mobileConnectionCore,
+        // but integration tests running on desktop without mobileConnection
+        // is testing this.
+      ]).then(function() {
+        this.startAPIHandlers();
+        return Promise.resolve();
+      }.bind(this)).then(function() {
+        return LazyLoader.load([
+          'js/download/download_manager.js',
+          'js/payment.js',
+          'js/identity.js',
+          'js/devtools/logshake.js',
+          'js/devtools/remote_debugger.js',
+          'js/devtools/developer_hud.js',
+          'shared/js/date_time_helper.js'
+        ]);
+      }.bind(this)).then(function() {
+        this.remoteDebugger = new RemoteDebugger();
+        this.developerHud = new DeveloperHud();
+        this.developerHud.start();
+      }.bind(this));
+    },
+
+    startAPIHandlers: function() {
       for (var api in this.REGISTRY) {
         this.debug('Detecting API: ' + api +
           ' and corresponding module: ' + this.REGISTRY[api]);
```

Comment:
If you follow the current system app convention:

1. literal object for those "stateless" features as Alive commented in this PR below
2. prototype + method + constructor for those "stateful" features
3. literal object could be "stateful" only when the context set it as a prototype

And:

4. we hate "Singleton" which usually use literal object with `this` and it keeps correct only because we usually use it with `addEventListener` which allows to bind the object as the `this`. It's a bad pattern that we once abandoned.

If this is true, it means the literal object with `this` for Base and extended module is creating a new convention beyond what we have.  I don't see why a convention change like this needn't comments. And if we all agree convention is important (think about those painful experiences from the "Singletons"), we should at least describe it as clear as possible.